.github/workflows/update-public-instances.yml

```
@@ -0,0 +1,48 @@
+name: Update public instances
+on:
+  push:
+    branches:
+      - instance-picker
+
+jobs:
+  run:
+    name: Update public instances
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout matrix.org source
+        uses: actions/checkout@v4
+        with:
+          path: 'morg'
+
+      - name: Checkout the public instance updater
+        uses: actions/checkout@v4
+        with:
+          repository: 'thibaultamartin/public-instances-updater'
+          path: 'piu'
+
+      - name: Update public instance data on the runner
+        run: |
+          cp morg/content/public-instances/instances.toml piu/instances.toml
+          cd piu && yarn && yarn run update && cd ..
+          cp piu/instances.toml morg/content/public-instances/instances.toml
+
+      - name: Commit and push data (within PR)
+        if: ${{ github.event_name == 'pull_request' }}
```

No need for `%{{` inside `if`

```suggestion
        if: github.event_name == 'pull_request'
```